bugsnag/middleware.py

```diff
@@ -104,6 +104,38 @@ def append(self, middleware):
         """
         self.stack.append(middleware)
 
+    def insert_before(self, middleware, target_class):
+        """
+        Adds a middleware to the stack in the position before
+        the target_class.
+        """
+        index = self._find_class_index(target_class)
+        if index >= 0:
+            self.stack.insert(index, middleware)
+        else:
+            self.append(middleware)
+
+    def insert_after(self, middleware, target_class):
+        """
+        Adds a middleware to the stack in the position after
+        the target_class.
+        """
+        index = self._find_class_index(target_class)
+        if index >= 0:
+            self.stack.insert(index + 1, middleware)
+        else:
+            self.append(middleware)
+
+    def _find_class_index(self, target_class):
+        """
+        Locates the index of the target class in the stack.
+        Returns -1 if not present.
+        """
+        for index, item in enumerate(self.stack):
+            if item.__name__ == target_class:
```

Comment:
It might be a better interface to have `insert_before` and `insert_after` take in actual class references/objects instead of strings since different classes can have the same `__name__` property.